raiden/utils/claim.py

```diff
@@ -249,8 +256,16 @@ def create_hub_claims(
     hub_address: Address,
     addresses: List[Address],
     token_amount: TokenAmount = DEFAULT_TOKEN_AMOUNT,
+    num_direct_channels: int = 0,
 ) -> Generator[Claim, None, None]:
     channels = []
+
+    # Open direct channels for node pairs, i.e. (0, 1), (2, 3), (4, 5)
+    # Nodes still are connected to the hub
+    for partner1, partner2 in zip(addresses[::2], addresses[1::2])[:num_direct_channels]:
```

Comment:
Subscript access doesn't work on `zip` (since it's a generator).

I'd instead use:
```
for partner1, partner2 in islice(zip(islice(addresses, 0, None, 2), islice(addresses, 1, None, 2)), num_direct_channels):
```

Reply:
I decided to just open all channels between the first X nodes:

```py
    # Open direct channels for node pairs, i.e. (0, 1), (1, 2), (2, 3)
    # Nodes still are connected to the hub
    for partner1, partner2 in islice(zip(addresses, addresses[1:]), num_direct_channels):